array-destructuring/exercise-2/exercise.js

```diff
@@ -70,3 +70,10 @@ let hogwarts = [
     occupation: "Teacher",
   },
 ];
+
+//let {firstName,lastName,house,pet,occupation} = hogwarts;
+hogwarts.filter(({firstName,lastName,house,pet,occupation }= hogwarts)=>{
+  house==="Gryffindor"?console.log(`${firstName} ${lastName}`):'';
+  occupation==="Teacher" && pet!== null?console.log(`--------------\n${firstName} ${lastName}`):'';
```

Comment:
This gives the right output but only because Albus Dumbledore appears at the end of the `hogwarts` list of people. It would be better to have two different filters here, and in fact this loop doesn't really filter as it would do the same if `forEach()` were used instead of `filter()`.

For example, we could do:

```js
hogwarts
  .filter(({ house }) => house === "Gryffindor")
  .forEach(({ firstName, lastName }) => {
    console.log(`${firstName} ${lastName}`);
  });

console.log("---------------");

hogwarts
  .filter(({ occupation, pet }) => occupation === "Teacher" && pet)
  .forEach(({ firstName, lastName }) => {
    console.log(`${firstName} ${lastName}`);
  });
```

Note how the `forEach()` call is the same each time - we separate the filtering out from printing out the names.